spartan/src/polycommitments/mod.rs

```diff
@@ -51,13 +51,9 @@ pub trait PolyCommitmentTrait<G: CurveGroup>:
   // this should be the commitment to the zero vector of length n
   fn zero(n: usize) -> Self;
 
-  fn into_affine(self) -> Vec<G::Affine> {
-    self.into().iter().map(|c| c.into_affine()).collect()
-  }
-
-  // for commitments in the form of a single field element (e.g., Zeromorph) this should return it
-  // for commitments unable to be representated as such (e.g., Hyrax) it should panic unimplemented!()
-  fn into_single(self) -> G;
+  // for commitments in the form of a single field element (e.g., Zeromorph) this should return Some(x)
```

Comment:
As per my previous comment, you can provide default implementation which returns `None`.

The name is misleading, I still suggest `into_affine` or even `try_into_affine` because of `Option`. We want this method for hashing, and protective coordinates aren't unique. No use for it otherwise.

Finally, consider adding something similar to the suggested doc comment in the code snippet, for example developers will be able to hover over this function and see what it does, instead of how it _should_ be implemented. (Meaning /// as well)